rust/lazy/src/lazy_config.rs

```diff
@@ -0,0 +1,163 @@
+use std::{
+    cmp::Ordering,
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader, Write},
+    path::Path,
+    vec,
+};
+
+use crate::error::{LazyError, Result};
+pub use crate::lazy_rule::*;
+
+#[derive(Debug)]
+pub struct LazyConfig {
+    rules: Vec<LazyRule>,
+}
+
+impl LazyConfig {
+    pub async fn load(reader: &mut impl BufRead) -> Result<Self> {
+        let mut rules: Vec<LazyRule> = vec![];
+
+        for line in reader.lines() {
+            let line_content = line?;
+            // empty lines
+            if line_content.trim().is_empty() {
+                continue;
+            }
+            // comments
+            if line_content.trim().starts_with('#') {
+                continue;
+            }
+            let rule: LazyRule = line_content.parse()?;
+            rules.push(rule);
+        }
+
+        // sort the rules by alphabetical order, and wildcard always at the top
+        rules.sort_by(|a, b| {
```

Comment:
Why would we need to sort these rules?  

Are we able to handle things like:


'*' smudge
'a/*' pointer
'a/b/*'  smudge
```
Here, if my rule file is like this, I would expect e.g. a/data.dat to be a pointer, but a/b/foo.bar to be smudged.  I'm not sure sorting them would preserve the precedence.

Reply:
Wildcards are not supported at the moment, but only in the scenario that a user defines something like `pointer a/data*`, as he wants to keep files under `a/` with prefix "data" as pointer files.

The exact behavior you want is supported, but defined in a different way:

```
smudge *
a pointer
a/b smudge
```

The reason to sort the rules by target path is that I can find the longest match first to implement the above behavior.

Reply:
Ah, ok, gotcha.  Interesting.  (Also, apologies -- for some reason I didn't see that notion document.)  If this is the consensus behavior, I'm down with it.  But I'm not convinced about the longest match behavior.